### PR TITLE
Source more ZIM metadata from WARC files

### LINKED ISSUE #357: Use `software` from `warcinfo` record to set `Scraper` metadata
Currently, scraper sets the `Scraper` ZIM metadata based on warc2zim version + an optional scraper suffix (usually passed by zimit with zimit version + browsertrix version).

This is not the optimal solution because WARC file already contain one (usually) `warcinfo` record which contains the browsertrix version in a `software` property.

We should prefer to use this information rather than fetching it indirectly in Zimit, especially since begining with https://github.com/openzim/zimit/issues/301 the WARC processed might come from a browsertrix version different than the one packaged with zimit. Even without this issue, it would allow to use any WARC harvester and still automatically have a proper `Scraper` metadata.

### LINKED ISSUE #358: Store `ContentDate` as metadata, based on WARC-Date
We should compute and store in a ZIM metadata the dates when the content has been fetched into the WARC.

This information is present in WARC records but lost once converted into a ZIM. This is a pity since it makes debugging ZIM issues even harder.

See https://github.com/openzim/overview/issues/40 for a proposition of standardization of this metadata, but we can move forward even if it is not yet standard.

### PR DESCRIPTION
**Pull request by benoit74**
Fix #358 
Fix #357 

Changes:
- in addition to changes needed by issues mentioned, we stop to filter warc records by type in the `iter_warc_records` generator since this is both confusing + needless ; selecting proper record type is the "using code" responsibility
and needs might vary

### CODE REVIEW

File: src/warc2zim/converter.py [hunk 1]
```diff
@@ -388,15 +407,46 @@ def run(self):
...
+            )
+            return
+        for warcfield in get_record_content(record).decode("UTF-8").splitlines():
+            name_and_value = warcfield.split(":")
+            if len(name_and_value) != 2:  # noqa: PLR2004
+                logger.warning(f"Unrecognized WARC field: {warcfield}")
+                continue
+            if name_and_value[0].strip().lower() != "software":
+                continue
+            self.warc_software = "," + name_and_value[1].strip()
```
Comment (rgaudin) [inline src/warc2zim/converter.py:427]: this is ugly. you should handle the concatenation properly when setting `Scraper`

File: src/warc2zim/converter.py [hunk 2]
```diff
@@ -388,15 +407,46 @@ def run(self):
...
+        if self.warc_software:
+            logger.debug("warc_software already set, ignoring warcinfo record")
+            return
+        if get_record_mime_type(record) != "application/warc-fields":
+            logger.warning(
+                f"Unsupported warcinfo record found: {get_record_mime_type(record)}"
+            )
+            return
+        for warcfield in get_record_content(record).decode("UTF-8").splitlines():
+            name_and_value = warcfield.split(":")
```
Comment (rgaudin) [inline src/warc2zim/converter.py:421]: I believe the following is as good (worst case is value contains additional stuff) yet clearer
```suggestion
            name, value = warcfield.split(":", 1)
            if name.strip().lower() != "software":
                continue
            self.warc_software = value.strip()
```

**Review by rgaudin — CHANGES_REQUESTED** [review]
Mostly OK.
I think that since there are ongoing discussions about storing this information and since there is no reader supporting it, we should use a safe name for now like `WARC-ContentDate` or `X-ContentDate` or something as for it not to interfere.

File: src/warc2zim/converter.py (hunk 1 above)
Comment (benoit74) [reply src/warc2zim/converter.py:427]: I have to admit I'm mostly ashamed, especially when I see the code setting `Scraper` metadata which is mostly as ugly.

**Review by rgaudin — APPROVED** [review]
LGTM

**Pull request merged**